Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_Signal.c

```diff
@@ -175,6 +175,13 @@ installSignalHandler()
                 sigaction(fatalSignals[i], &g_previousSignalHandlers[i], NULL);
             }
             goto failed;
+        } else {
+            // The previous handler was `SIG_IGN` -- restore the original handler so
+            // we don't override the `SIG_IGN` and report a crash when the application
```

Comment:
Instead of restoring the original signal handler here, can't we check above if the handler is `SIG_IGN`?

Reply:
We have to make the `sigaction` call since that's what will actually store the old signal handler into `g_previousSignalHandlers[i]`. Before that call the entire array is just an empty vector.

See the [sigaction](https://man7.org/linux/man-pages/man2/sigaction.2.html) documentation:

> If act is non-NULL, the new action for signal signum is installed
       from act.  If oldact is non-NULL, the previous action is saved in
       oldact.